src/main/java/me/zeroeightsix/fiber/serialization/Serializer.java

```diff
@@ -11,14 +11,30 @@
 /**
  * A {@code Serializer} serializes and deserializes data from a certain format into and from {@code Node}s
  *
+ * @param <T> the type of serialized objects processed by this serializer
  * @see Node
  * @see JanksonSerializer
  */
-public interface Serializer {
+public interface Serializer<T> {
 
     Identifier getIdentifier();
 
     void deserialize(Node node, InputStream stream) throws FiberException, IOException;
+
+    /**
+     * Deserializes an intermediate element into a config tree.
+     *
+     * <p> If the serialized tree has elements not present in the actual node,
+     * they will be collected as a separate data structure and returned.
+     *
+     * @param node the node to deserialize
+     * @param element the serialized data for the node
+     * @return the unprocessed elements, in the same format as the input data
+     */
+    T deserialize(Node node, T element) throws FiberException;
+
     void serialize(Node node, OutputStream stream) throws FiberException, IOException;
 
+    T serialize(Node node) throws FiberException;
```

Comment:
Because we're no longer including the 'leftovers' in the IR tree, it might be appealing to allow you to pass the leftovers back to the serializer when serializing: that way users don't have to figure out how to shove the leftovers back into the resulting configuration.

In case we don't care about preserving unprocessed data, that argument should be `@Nullable` as well.

Reply:
Good idea.